src/ibrowse_http_client.erl

```diff
@@ -268,10 +269,46 @@ terminate(_Reason, State) ->
 code_change(_OldVsn, State, _Extra) ->
     {ok, State}.
 
+
+%%--------------------------------------------------------------------
+%% Function: format_status/3
+%% Purpose: Clean process state before logging
+%% Returns: key value list
+%%--------------------------------------------------------------------
+format_status(_Opt, [_PDict, State]) ->
+    #state{
+        reqs=Reqs,
+        reply_buffer=ReplyBuf,
+        recvd_headers=RCVDHeaders,
+        raw_headers=RawHeaders,
+        chunk_size_buffer=ChunkSizeBuf,
+        cur_req=Request
+    } = State,
+    ScrubbedReq = Request#request{url=url_strip_password(Request#request.url)},
+    Scrubbed = State#state{
+        socks5_user=nil,
+        socks5_password=nil,
+        reqs=queue:len(Reqs),
```

Comment:
I see we reference the queue here: https://github.com/apache/couchdb-ibrowse/blob/f4fd3f11747fd9b03fe64c94c7288653fd5f7320/src/ibrowse_http_client.erl#L507, would changing this to an integer cause a badargs?

Reply:
No it will not cause problems because the result of `format_status/2` is only used for logging purposes and doesn't modify actual state. See here:

1. [`gen_server:format_status/2`](https://github.com/erlang/otp/blob/master/lib/stdlib/src/gen_server.erl#L1201) is called from [`error_info/7`](https://github.com/erlang/otp/blob/master/lib/stdlib/src/gen_server.erl#L932:L945) which return `ok`. I.e. result is not used.
2. `gen_server:format_status/2` dispatches to `gen_server:format_status/4` [here](https://github.com/erlang/otp/blob/master/lib/stdlib/src/gen_server.erl#L1201).
3. `gen_server:format_status/2` calls `Module:format_status/2` if it is defined from here https://github.com/erlang/otp/blob/master/lib/stdlib/src/gen_server.erl#L1225:L1233

Reply:
Yup, I see  it now. Since this is all for logging, should we change the newly calculated numerical values into a binary with more meaningful info? 

<<"Reqs Size...>>
<<"ReqHeader Size..>
etc..?

Reply:
or we could do `reqs={queue_size, queue:len(Reqs)}`.

Reply:
ya, that could work